test/helpers/webhook.go

```diff
@@ -79,7 +79,7 @@ func appendWebhookConfiguration(configyamlFile []byte, tag string) ([]*v1.Mutati
 
 func initializeWebhookInEnvironment() {
 	// Get the root of the current file to use in CRD paths.
-	_, filename, _, _ := goruntime.Caller(0) //nolint
+	_, filename, _, _ := goruntime.Caller(0) //nolint:dogsled
```

Comment:
```suggestion
	_, filename, _, ok := goruntime.Caller(0)
	if !ok {
		klog.Fatalf("Failed to get information for current file from runtime")
	}
```

And elsewhere for the dogsled would mean we don't have to ignore this and we're handling an error earlier. IMO this is an improvement over ignoring the linter, and we can make the same change in core CAPI.

Reply:
Sounds good!

Reply:
Noted! 😄